src/types/service.ts

```diff
@@ -33,6 +34,6 @@ export function toAppService(from: ServicePublic): Service {
     organizationName: from.organizationName,
     serviceId: from.serviceId,
     serviceName: from.serviceName,
-    version: from.version || 0
+    version: from.version as NonNegativeNumber
```

Comment:
for `0`  above you are allowed to cast to `NonNegativeNumber` since you're sure that `0` is a `NonNegativeNumber`

in this case though, you can't cast, since `from` comes from an eternal source, you have to validate it first